### DISCUSSION TITLE
Why do we define the COLLECTION_CREATOR and MINT_FEE_COLLECTOR as manualy in nft-minting template?

### DISCUSSION THREAD

**Discussion by CryptainGolD** [aptos-labs/aptos-developer-discussions#519] [General]
In nft minting template, we define above two addresses as manualy in `.env`.
Let's assume I defined them as my wallet address and published contract with my wallet private key.
Once other users create collections with there wallet, then all collection's creator will be my wallet account. right?
And next, once `user A`(not me) has created a collection on my contract, and `user B` is minting that collection, then also, all minting fee will go to my wallet.
So I want to make creator and minting fee collector as dynamically. Guide me on that template with code please I am a beginner at move.

**Comment by Pcharlesme** [comment]
@Cryptain-Harlock 
Here’s how you can modify your contract to achieve dynamic creator and minting fee handling.

Step 1: Define the Collection Creator

Add a mapping to store the creator address for each collection.
let me  drop some example

**Comment by Pcharlesme** [comment]
During the minting process, reference the creator stored in the Collection struct and transfer the fee.

`public fun mint_nft(
    minter: &signer,
    collection_name: vector<u8>,
    amount: u64,
) acquires CollectionStore {
    let minter_address = signer::address_of(minter);

    // Verify that the collection exists
    let collection_store = borrow_global<CollectionStore>(minter_address);
    let collection = table::borrow(&collection_store.collections, &collection_name);

    // Verify that the minter has enough funds
    assert!(coin::balance(&minter) >= collection.minting_fee * amount, 1);

    // Transfer minting fees to the collection creator
    let creator_address = collection.creator;
    coin::transfer(minter, creator_address, collection.minting_fee * amount);

    // Mint the NFT (not implemented here, would involve adding NFT to minter's account)
}
`

**Comment by Pcharlesme** [comment]
mapping to store the creator address for each collection.

`module NFT {
    use aptos_framework::coin;
    use aptos_framework::aptos_account;
    use aptos_framework::event;
    use aptos_framework::signer;

    // Struct to represent a collection
    struct Collection has key {
        creator: address, // Store the creator address dynamically
        name: vector<u8>,
        description: vector<u8>,
        minting_fee: u64, // Minting fee in smallest currency units
    }

    // Mapping from collection name to Collection struct
    struct CollectionStore has key {
        collections: table::Table<vector<u8>, Collection>, // Collection name -> Collection details
    }

    // Resource for tracking collections
    public fun create_collection(
        account: &signer,
        name: vector<u8>,
        description: vector<u8>,
        minting_fee: u64,
    ) {
        let creator = signer::address_of(account);

        // Initialize a collection
        let collection = Collection {
            creator,
            name: name.clone(),
            description,
            minting_fee,
        };

        // Save the collection in the account's collection store
        if (!exists<CollectionStore>(creator)) {
            move_to(account, CollectionStore {
                collections: table::new(),
            });
        }

        let collection_store = borrow_global_mut<CollectionStore>(creator);
        table::add(&mut collection_store.collections, name, collection);
    }
}

`
